src/interface.jl

```diff
@@ -468,3 +469,24 @@ function buffered_operate_fallback!(
 ) where {N}
     return mutable_buffered_operate!(buffer, op, args...)
 end
+
+# For most types, `dot(b, c) = adjoint(b) * c`.
```

Comment:
Are there any types for which this isn't true?

Reply:
Array, currently, dot would not work as expected wirh array of arrays

Reply:
This hasn't changed with this PR so let's leave it for another one. 
Opened an issue: https://github.com/jump-dev/MutableArithmetics.jl/issues/92